api-aws-lambda/src/main/kotlin/org/coepi/api/v4/TCNCloudAPIHandler.kt

```diff
@@ -53,13 +65,11 @@ class TCNCloudAPIHandler: RequestHandler<APIGatewayProxyRequestEvent, APIGateway
         var body: String?
 
         try {
-            val queryParams = parseQueryParameters(input)
+            val (maybeDate, maybeInterval) = parseQueryParameters(input)
 
-            val date = queryParams.first.orElse(LocalDate.now())
-            val intervalNumber = queryParams.second.orElse(generateIntervalForCurrentTime())
+            logger.info("Querying reports with date: $maybeDate and intervalNumber: $maybeInterval")
 
-            logger.info("Querying reports with date: $date and intervalNumber: $intervalNumber")
-            val reports = reportsDao.queryReports(date, intervalNumber)
+            val reports = reportService.getReports(maybeDate.orNull(), maybeInterval.orNull())
                     .map {
                         record -> Base64.getEncoder().encodeToString(record.report)
```

Comment:
can be moved to the core. Even the exception -> body, statuscode mapping can be moved to the core. The core handler can return an object of type HttpResponse, and all the lambda handler needs to do then is convert a standard response to APIGatewayProxyResponse. Then the web server front-end won't need to replicate this logic